lib/src/svg_parser.dart

```diff
@@ -12,7 +12,9 @@ import 'vector_drawable.dart';
 
 /// An SVG Shape element that will be drawn to the canvas.
 class DrawableSvgShape extends DrawableShape {
-  const DrawableSvgShape(Path path, DrawableStyle style) : super(path, style);
+  const DrawableSvgShape(Path path, DrawableStyle style, this.transform) : super(path, style);
+
+  final Matrix4 transform;
```

Comment:
Not clear on why this is changing.  The code as it is is slightly more optimized for draw operations (paths are pre-transformed).  It doesn't look like your patch is taking advantage of the new logic anyway, unless I'm missing something.

Reply:
The problem is that, in the code before this fix, shaders were not transformed, just the shapes.
To give you an example, if you have a Rect and you paint it with a linear gradient from bottom-left to top-right and you have a 90 deg rotate transform on it's element, you should see the gradient from top-left to bottom-right, but in the previous code that wasn't the case.

The new code is taking advantage of that property on the new overridden `draw` method of `DrawableSvgShape`, which is performing a `save/transform/draw/restore` to apply the transform in case it is set.

If I correctly understood how transforms are applied when drawing on a canvas, I don't think that would be a major performance overhead.

Reply:
Ahhh I see now.  This makes sense then.